### PR TITLE
Improve selection of compiler & linker for Windows and Intel

### LINKED ISSUE #859: testing new Windows configuration
Dear Pyccel team,
I'm working on a Windows pc, and apparently I have all the pyccel-requirements already installed. Everything is easily provided for free by:
- Visual Studio Community (2017 or 2019), with Python 3.7 64 bit [Python Software Fundation]   ... (not Conga, or Anaconda or others...);
- Intel oneAPI Base Toolkit (compilers, pre-optimized libraries (MKL included), analysis tools,....)
- Intel oneAPI HPC Toolkit (C++,Fortran,OpenMP,MPI,...)

I don't know if you provide support for testing Pyccel on new platforms, just let me know and eventually close this issue. In principle, if I had info about the meaning of the error-messages, I could try to fix something...

Now: I installed Pyccel with pip and I'm running sequentially the pyccel-tests  provided in the `tests\run_tests.bat`. I passed all the test until the following:
```
python -m pytest -rx --ignore=symbolic --ignore=ndarrays -v --maxfail=1 -v ../tests/epyccel -m "not parallel"
```
when I get the following error message (I append only the first and last lines):

```python
==================================================================================================================== test session starts ====================================================================================================================
platform win32 -- Python 3.7.8, pytest-6.2.3, py-1.10.0, pluggy-0.13.1 -- C:\Users\myuser\AppData\Local\Programs\Python\Python37\python.exe
cachedir: .pytest_cache
rootdir: C:\Users\myuser\Documents\00_Tools\Pyccel, configfile: pytest.ini
plugins: forked-1.3.0, xdist-2.2.1
collected 2659 items / 13 deselected / 2646 selected

epyccel\test_array_as_func_args.py::test_array_int_1d_scalar_add[fortran] FAILED                                                                                                                                                                       [  0%]

========================================================================================================================= FAILURES ==========================================================================================================================
___________________________________________________________________________________________________________ test_array_int_1d_scalar_add[fortran] ___________________________________________________________________________________________________________

language = 'fortran'

    def test_array_int_1d_scalar_add(language):
        @types( 'int8[:]' , 'int8' , 'int')
        @types( 'int16[:]', 'int16', 'int')
        @types( 'int32[:]', 'int32', 'int')
        @types( 'int64[:]', 'int64', 'int')
        def array_int_1d_scalar_add( x, a, x_len ):
            for i in range(x_len):
                x[i] += a
        f1 = array_int_1d_scalar_add
>       f2 = epyccel(f1, language=language)

epyccel\test_array_as_func_args.py:24:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
C:\Users\myuser\AppData\Local\Programs\Python\Python37\lib\site-packages\pyccel\epyccel.py:335: in epyccel
    mod, fun = epyccel_seq( python_function_or_module, **kwargs )
C:\Users\myuser\AppData\Local\Programs\Python\Python37\lib\site-packages\pyccel\epyccel.py:180: in epyccel_seq
    output_name = module_name)
C:\Users\myuser\AppData\Local\Programs\Python\Python37\lib\site-packages\pyccel\codegen\pipeline.py:458: in execute_pyccel
    verbose)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

codegen = <pyccel.codegen.codegen.Codegen object at 0x00000251FF5E8B48>, language = 'fortran', pyccel_dirpath = 'C:\\Users\\myuser\\Documents\\00_Tools\\Pyccel\\tests\\__epyccel__\\__pyccel__', compiler = 'gfortran', mpi_compiler = None
accelerator = None
dep_mods = ('C:\\Users\\myuser\\Documents\\00_Tools\\Pyccel\\tests\\__epyccel__\\__pyccel__\\bind_c_mod_a9gxdwww2u2p_a9gxdwww2u2p', 'C:\\Users\\myuser\\Documents\\00_Tools\\Pyccel\\tests\\__epyccel__\\__pyccel__\\mod_a9gxdwww2u2p_a9gxdwww2u2p')
libs = [], libdirs = [], includes = ['C:\\Users\\myuser\\Documents\\00_Tools\\Pyccel\\tests\\__epyccel__\\__pyccel__', 'C:\\Users\\myuser\\AppData\\Local\\Programs\\Python\\Python37\\lib\\site-packages\\numpy\\core\\include']
flags = ' -O3 -fPIC  -I"C:\\Users\\myuser\\Documents\\00_Tools\\Pyccel\\tests\\__epyccel__\\__pyccel__"', sharedlib_modname = 'mod_a9gxdwww2u2p_a9gxdwww2u2p', verbose = False

    def create_shared_library(codegen,
                              language,
                              pyccel_dirpath,
                              compiler,
                              mpi_compiler,
                              accelerator,
                              dep_mods,
                              libs,
                              libdirs,
                              includes='',
                              flags = '',
                              sharedlib_modname=None,
                              verbose = False):
... code lines
... code lines
... code lines
... code lines
            if verbose:
                print(out)
            if p.returncode != 0:
                err_msg = "Failed to build module"
                if verbose:
                    err_msg += "\n" + err
>               raise RuntimeError(err_msg)
E               RuntimeError: Failed to build module

C:\Users\myuser\AppData\Local\Programs\Python\Python37\lib\site-packages\pyccel\codegen\python_wrapper.py:148: RuntimeError
------------------------------------------------------------------------------------------------------------------- Captured stdout call --------------------------------------------------------------------------------------------------------------------

ERROR at shared library generation stage
----------------------------------------------------------------------------------------------------------------- Captured stdout teardown ------------------------------------------------------------------------------------------------------------------
Tearing down!
```
this message does not help me to fix the problem, should I run other basic tests before this?

many thanks and best regards

### PR DESCRIPTION
**Pull request by EmilyBourne**
Fix #859:
- Remove need for system-wide compiler choice on Windows
- Compile and link wrapper with Intel when compiling generated sources with Intel

On Windows provide the argument to the `setup_X.py` file when the compiler is from the intel or gcc families, to avoid the need for the hacky modification of the `distutils.cfg` file.  Provide 2 possible names for the gfortran library on Windows. Set environment variables in the correct place to correctly compile and link files with Intel.

### CODE REVIEW

File: pyccel/codegen/utilities.py [hunk 1]
```diff
@@ -177,14 +177,16 @@ def get_gfortran_library_dir():
...
-            sys.path.insert(0, lib_dir)
+        file_name_list = ['libgfortran.a']
+
+    for file_name in file_name_list:
+        file_location = subprocess.check_output([shutil.which('gfortran'), '-print-file-name='+file_name],
+                universal_newlines = True)
+        lib_dir = os.path.abspath(os.path.dirname(file_location))
+        if lib_dir != os.getcwd():
+            if lib_dir not in sys.path:
+                # Add to sytem path
```
Comment (yguclu) [inline pyccel/codegen/utilities.py:190]: Typo (I know, it was here before): sytem -> system

File: pyccel/codegen/python_wrapper.py [hunk 2]
```diff
@@ -127,6 +127,8 @@ def create_shared_library(codegen,
 
         setup_filename = os.path.join(pyccel_dirpath, setup_filename)
         cmd = [sys.executable, setup_filename, "build"]
+        if sys.platform == 'win32' and compiler in ('gfortran','gcc','ifort','icc'):
+            cmd+=['--compiler=mingw32']
```
Comment (yguclu) [inline pyccel/codegen/python_wrapper.py:131]: A bit of whitespace around `+=` is not forbidden :wink:

Comment (EmilyBourne) [reply pyccel/codegen/python_wrapper.py]: Fixed in 8bd329d

File: pyccel/codegen/utilities.py (hunk 1 above)
Comment (EmilyBourne) [reply pyccel/codegen/utilities.py]: Fixed in 8bd329d

**Review by yguclu — APPROVED** [review]

**Pull request merged**